lib/contentful/client.rb

```diff
@@ -296,6 +298,16 @@ def sync(options = { initial: true })
 
     private
 
+    def ensure_select_is_doable(query)
```

Comment:
I prefer the mutating method in this case, it simplifies the code, also the result query object is passed immediately, therefore a non-mutating solution doesn't provide any benefit.